io/io/src/TMemFile.cxx

```diff
@@ -55,10 +55,20 @@ TMemFile::TMemBlock::TMemBlock() : fPrevious(0), fNext(0), fBuffer(0), fSize(0)
 TMemFile::TMemBlock::TMemBlock(Long64_t size, TMemBlock *previous) :
    fPrevious(previous), fNext(0), fBuffer(0), fSize(0)
 {
-   fBuffer = new UChar_t[size];
-   fSize = size;
+   // size will be -1 when copying an existing buffer into fBuffer.
+   if (size != -1) {
+      fBuffer = new UChar_t[size];
+      fSize = size;
+   }
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor not allocating the memory buffer, for external ownership.
+
+TMemFile::TMemBlock::TMemBlock(UChar_t* data, Long64_t size) :
```

Comment:
Why are both this constructor and the (size==-1) in the previous constructor needed?

Reply:
This one is for external memory that we do not copy, the size==-1 is for memory that we do copy. See the comment "   // size will be -1 when copying an existing buffer into fBuffer."

Reply:
Then this convention **must** be expressed in the constructor's documentation.

Reply:
See 3d71859f73cea5ac4e94f087946827a4d3d84716 !